### ISSUE TITLE
non standard characters in username

### ISSUE THREAD

**Issue by UseDotCom** [UseDotCom/lighthouse#509]
Created by **[usedotcom](https://github.com/usedotcom)**  at 14:53Z on 05/21/2013 *(via [Lighthouse](http://use-com.lighthouseapp.com/projects/103799/tickets/90))*

- - - -

The sign up form allows non standard characters in the username (such as semi-colons). These cause errors when trying to access your profile page and within the iframes on the My Account page. Need to restrict allowable characters to alpha numeric plus - and _. See attached screenshot.

- - - -

Lighthouse ticket URL: http://use-com.lighthouseapp.com/projects/103799/tickets/90

Attachments:

* https://use-com.lighthouseapp.com/attachments/3529292/non-standard-characters-username.png
* https://use-com.lighthouseapp.com/attachments/3570331/special-email-list.txt
* https://use-com.lighthouseapp.com/attachments/3570332/email1.txt

Was assigned to **[](https://github.com/)**

**Comment by UseDotCom** [comment]
By **[usedotcom](https://github.com/usedotcom)** at 18:06Z on 05/20/2013 *(via [Lighthouse](http://use-com.lighthouseapp.com/projects/103799/tickets/90))*

- - - -

Cannot log in if your current username contains special characters. Preferred fix is to allow existing usernames that contain these characters and prohibit them in new usernames. If this is not reasonably feasible, I need a query/list of all usernames that contain special characters and their email addresses so we can get them to change their usernames.

**Comment by UseDotCom** [comment]
By **Steve** at 21:32Z on 05/20/2013 *(via [Lighthouse](http://use-com.lighthouseapp.com/projects/103799/tickets/90))*

- - - -

Fixed.

Requirements of the user name now match that of RFC-1123 (Requirements for Internet Hosts). That's 'A' through 'Z', '0' through '9' and the hyphen ('-'). Furthermore, while hyphens may be used in the username, they must not begin or end with a hyphen.

Email of non conforming users have been sent to Val.

Please check functionality.

**Comment by UseDotCom** [comment]
By **[usedotcom](https://github.com/usedotcom)** at 22:05Z on 05/20/2013 *(via [Lighthouse](http://use-com.lighthouseapp.com/projects/103799/tickets/90))*

- - - -

Sign up is still allowing # in the username box. It does give an Oops  
message upon submission. Other invalid characters give the red "Invalid  
characters" warning when entered
 
 
 
In a message dated 5/20/2013 5:34:29 P.M. Eastern Daylight Time,

**Comment by UseDotCom** [comment]
By **[usedotcom](https://github.com/usedotcom)** at 14:53Z on 05/21/2013 *(via [Lighthouse](http://use-com.lighthouseapp.com/projects/103799/tickets/90))*

- - - -

I just discovered that, usernames with special characters (_) and (.) can only log in to their account on the photo side.  Regardless of which side they log in on, they are not logged in on the profile side but are logged on their photo side. So it will be necessary to send out a dynamic mass email sooner rather than later.

Attached:

stock email - this has a place for the username. This is the only dynamic portion of the email
email list - I have removed those that have already been contacted.

Send from cust.support@use.com. BCC kellyj@priceside.com

Subject: Use.com Account Update
